packages/android_intent/android/src/main/java/dev/flutter/plugins/androidintent/IntentSender.java

```diff
@@ -0,0 +1,116 @@
+package dev.flutter.plugins.androidintent;
+
+import android.app.Activity;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Log;
+import androidx.annotation.Nullable;
+
+/** Forms and launches intents. */
+public class IntentSender {
+  private static final String TAG = "IntentSender";
+
+  private @Nullable Activity activity;
```

Comment:
Should `activity` be a `WeakReference` as precaution? No opinion on my end, just a thought.

Reply:
I don't think so, but I may be missing something. This shouldn't be leaking because we're always updating the state from the `onDetatched...` callbacks, right? If those handlers aren't totally trustworthy I feel like that's a separate bug that we should find and fix rather than trying to guard against here.

Reply:
Yeah this should be good.